packages/client/src/store/CallState.ts

```diff
@@ -376,33 +372,60 @@ export class CallState {
       shareReplay({ bufferSize: 1, refCount: true }),
     );
 
+    // dates
+    this.createdAt$ = this.createdAtSubject.asObservable();
+    this.endedAt$ = this.endedAtSubject.asObservable();
+    this.startsAt$ = this.startsAtSubject.asObservable();
     this.startedAt$ = this.startedAtSubject.asObservable();
-    this.participantCount$ = this.participantCountSubject.asObservable();
-    this.anonymousParticipantCount$ =
-      this.anonymousParticipantCountSubject.asObservable();
+    this.updatedAt$ = this.updatedAtSubject.asObservable();
 
     this.callStatsReport$ = this.callStatsReportSubject.asObservable();
     this.members$ = this.membersSubject.asObservable();
-    this.ownCapabilities$ = this.ownCapabilitiesSubject.asObservable();
-    this.callingState$ = this.callingStateSubject.asObservable();
 
-    this.backstage$ = this.backstageSubject.asObservable();
-    this.blockedUserIds$ = this.blockedUserIdsSubject.asObservable();
-    this.createdAt$ = this.createdAtSubject.asObservable();
-    this.endedAt$ = this.endedAtSubject.asObservable();
-    this.startsAt$ = this.startsAtSubject.asObservable();
-    this.updatedAt$ = this.updatedAtSubject.asObservable();
+    // complex objects should work as streams of data
     this.createdBy$ = this.createdBySubject.asObservable();
     this.custom$ = this.customSubject.asObservable();
     this.egress$ = this.egressSubject.asObservable();
     this.ingress$ = this.ingressSubject.asObservable();
-    this.recording$ = this.recordingSubject.asObservable();
     this.session$ = this.sessionSubject.asObservable();
     this.settings$ = this.settingsSubject.asObservable();
-    this.transcribing$ = this.transcribingSubject.asObservable();
     this.endedBy$ = this.endedBySubject.asObservable();
     this.thumbnails$ = this.thumbnailsSubject.asObservable();
 
+    /**
+     * Performs shallow comparison of two arrays.
+     * Expects primitive values: [1, 2, 3] is equal to [2, 1, 3].
+     */
+    const isShallowEqual = <T>(a: Array<T>, b: Array<T>): boolean => {
+      if (a.length !== b.length) return false;
+      for (let i = 0; i < a.length; i++) {
+        if (b.indexOf(a[i]) === -1) return false;
```

Comment:
a = [1, 2, 2]
b = [1, 2, 3]

Will return true.

I suppose in our case all values are unique, but worth looking into it.

Reply:
Good point. Let me see how to fix this efficiently :)

Reply:
Fixed with: [affbc30](https://github.com/GetStream/stream-video-js/pull/1330/commits/affbc3003123e492cbde65c735ed6c390e22c10b)